Практика №9. Разработка с помощью PyQt5

WutenderBorsch opened the pull request:
Вариант 2.

1. Написать приложение, выполняющее построение синусоиды. Период синусоиды должен задаваться с помощью слайдера (QSlider).
2. Написать приложение, выполняющее просмотр изображений из указанного каталога. Должна быть возможность переходить от изображения к изображению с помощью кнопок «Влево» и «Вправо».

WutenderBorsch committed:
Практика №9. Разработка с помощью PyQt5

Вариант 2.

1. Написать приложение, выполняющее построение синусоиды. Период синусоиды должен задаваться с помощью слайдера (QSlider).
2. Написать приложение, выполняющее просмотр изображений из указанного каталога. Должна быть возможность переходить от изображения к изображению с помощью кнопок «Влево» и «Вправо».

silentonni commented:
Всё хорошо всё отлично!

silentonni commented on 1. Построение синусоиды/main.py:
```diff
@@ -0,0 +1,42 @@
+
+import sys
+
+class Window(QDialog):
```
Всё хорошо всё отлично!

silentonni commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,80 @@
+    i.x += 1
+    pixmap = QPixmap(str(images[i.x]))
+    lable.setPixmap(pixmap)
+    return
```
Можно объединить в одну строку: return lable.setPixmap(pixmap).

silentonni commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,80 @@
+
+    pixmap = QPixmap(str(images[i.x]))
+    lable.setPixmap(pixmap)
+    return
```
Можно объединить в одну строку: return lable.setPixmap(pixmap).

WutenderBorsch committed:
Update main.py

stanislavlyalin commented on 1. Построение синусоиды/main.py:
```diff
@@ -0,0 +1,42 @@
+        self.horizontalSlider.valueChanged.connect(self.widthChanged)
+        self.canvas.draw()
+
+    def foo(self, value):
```
Метод не используется. Удалить

stanislavlyalin commented on 1. Построение синусоиды/main.py:
```diff
@@ -0,0 +1,42 @@
+class Window(QDialog):
+    def __init__(self):
+        super().__init__()
+        uic.loadUi('untitled.ui', self)
```
`untitled.ui` - название не удачное. Дать осмысленное имя файлу

stanislavlyalin commented on 1. Построение синусоиды/main.py:
```diff
@@ -0,0 +1,42 @@
+        self.ax.plot(x, y, '-')
+        self.canvas.draw()
+
+app = QApplication(sys.argv)
+ex = Window()
+ex.show()
+sys.exit(app.exec_())
```
Лучше обернуть так:
```python
if __name__ == '__main__':
    app = QApplication(sys.argv)
    ex = Window()
    ex.show()
    sys.exit(app.exec_())

stanislavlyalin commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,78 @@
+from PyQt5.QtGui import *
+from PyQt5.QtWidgets import *
+
+app = QApplication(sys.argv)
+w = QWidget()
```
Этот код тоже лучше обернуть, как я показал выше

stanislavlyalin commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,78 @@
+app = QApplication(sys.argv)
+w = QWidget()
+
+patch = QFileDialog.getExistingDirectory(None, "Select folder", ".")
+
+if patch == '':
+    quit()
+
+files = os.listdir(patch)
+
+images = []
+
+for x in files:
+    if x.endswith('.png'):
+        images.append(patch + "\\" + x)
+
+if len(images) == 0:
+    quit()
```
Сделать отдельный класс для окна приложения. Открытие каталога, поиск файлов, построение списка изображений делать в конструкторе класса

stanislavlyalin commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,78 @@
+if len(images) == 0:
+    quit()
+
+lable = QLabel(w)
```
Наверное, всё-таки, `label`? :)

stanislavlyalin commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,78 @@
+if len(images) == 0:
+    quit()
+
+lable = QLabel(w)
+
+pixmap = QPixmap(str(images[0]))
+
+lable.setPixmap(pixmap)
+lable.setMaximumWidth(500)
+lable.setMaximumHeight(500)
```
Это тоже можно перенести в конструктор класса

stanislavlyalin commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,78 @@
+lable.setMaximumHeight(500)
+
+
+class Container:
+    def __init__(self, x):
+        self.x = x
+
+def Left(i):
+    if i.x == 0:
+        return
+
+    i.x -= 1
+
+    pixmap = QPixmap(str(images[i.x]))
+    return lable.setPixmap(pixmap)
+
+def Right(i):
+    if i.x == len(images) - 1:
+        return
+
+    i.x += 1
+    pixmap = QPixmap(str(images[i.x]))
+    return lable.setPixmap(pixmap)
```
Методы `left` и `right` должны быть в том же классе окна, о котором я написал выше

stanislavlyalin commented on 2. Просмотр изображений/main.py:
```diff
@@ -0,0 +1,78 @@
+
+    x = Container(0)
+
+    left.clicked.connect(lambda: Left(x))
+    fight.clicked.connect(lambda: Right(x))
```
Связывание кнопок и действий должно быть в конструкторе класса

WutenderBorsch committed:
Исправила недочеты

Исправила 3 файла:
1. Построение синусоиды/main.py
2. Файл View
3. Просмотр изображений/main.py
